frontend/packages/topology/src/data-transforms/DataModelProvider.tsx

```diff
@@ -19,6 +21,26 @@ interface DataModelProviderProps {
   children?: React.ReactNode;
 }
 
+const flattenResource = (
+  namespace: string,
+  model?: ExtensionK8sGroupKindModel,
+  opts = {} as Partial<WatchK8sResource>,
+) => {
+  if (!model) {
+    return { namespace, ...opts };
+  }
+
+  if (model?.version) {
+    const extensionReference = referenceForExtensionModel(model); // requires model.version
+    return { namespace, kind: extensionReference, ...opts };
+  }
+
+  // If can't find reference for an extention model, fall back to internal reference
+  const internalModel = modelForGroupKind(model.group, model.kind); // Return null for CRDs
```

Comment:
Why aren't we using the API version?

Reply:
I don't know how to get the API version at this point, `modelForGroupKind` is using the `store` to get it.
@spadgett what am I missing ?

Reply:
@spadgett help, what am I missing ?

Reply:
@spadgett these models are fetched form extension contributed via diff packages for topology

https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/extensions/topology.ts#L36-L54

where resources have the model with type `ExtensionK8sGroupKindModel`, where version is optional but kind group is required.

https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/api/common-types.ts#L21-L25

that is why `modelForGroupKind` is used here to get models with group and kind to get models and existing logic as well does that

https://github.com/openshift/console/blob/master/frontend/packages/topology/src/data-transforms/DataModelProvider.tsx#L37

Reply:
Hi @invincibleJai. Thanks. I'm not familiar with the topology extension points, but isn't the API version important if the extension needs to work with the data? Why was it left out?

We can make this a follow up discussion to unblock the bug fix.

Reply:
I am not very sure about the thought process while designing these extension points for topology but what I could think of is the reason to have this flexibility would be in topology we visualize resources as diff shapes uniquely by group and kind. Packages will provide a list of resources with the group and kind we topology is interested in, as in the case of operators at times they support multiple versions for the same resource. So based on group and kind can fetch the model for the preferred version if CRD has one or the latest one with `modelForGroupKind`. 

cc @christianvogt  can add more to it.

Reply:
> We can make this a follow up discussion to unblock the bug fix.

This bug affect users, unblocking the bug and creating a follow up discussion makes a lot of sense to me too.

Reply:
Thanks Sam, have logged one here https://bugzilla.redhat.com/show_bug.cgi?id=2039381

@yaacov based on the original ticket, have created a new one but to reproduce team would need help to get `kubevirt CI cluster` and even the issue was intermittent https://github.com/openshift/console/pull/10272#issuecomment-962907048 

so if there are steps to reproduce consistently or any way to reproduce it on another cluster, pls add a comment on the ticket.

Reply:
thanks, added @gouyang to the new bugzilla.

Reply:
> OK I don't want to block the bug any longer. Let's go ahead with the fix. Can we open a bug to track the follow on issue?

@spadgett hi, do I need to change something before you can lgtm it ?